#!/bin/sh

# Environment setup
QUAST_CODE_DIR="Apps/quast-5.2.0"
QUAST_OUTPUT_DIR="quast_results/"
GENOME_INDICES_OUTPUT_DIR="genome_indices"
GENOME_FASTA_DIR="lamprey_genome/Petromyzon_marinus.Pmarinus_7.0.dna.toplevel_13_02_2024.fa"
GENOME_ANNOTATION_DIR="lamprey_genome_annotations/Petromyzon_marinus.Pmarinus_7.0.112.gtf"
RNA1_DIR="ERR7132687_RNA.fastq"
RNA2_DIR="ERR7132689_RNA.fastq"
RNA3_DIR="ERR7132699_RNA.fastq"
RNA4_DIR="ERR7132707_RNA.fastq"
RNA1_TRIMMED_FIRST_DIR="ERR7132687_RNA_trimmed.fastq"
RNA2_TRIMMED_FIRST_DIR="ERR7132689_RNA_trimmed.fastq"
RNA3_TRIMMED_FIRST_DIR="ERR7132699_RNA_trimmed.fastq"
RNA4_TRIMMED_FIRST_DIR="ERR7132707_RNA_trimmed.fastq"
RNA1_TRIMMED_SECOND_DIR="ERR7132687_RNA_trimmed_second.fastq"
RNA2_TRIMMED_SECOND_DIR="ERR7132689_RNA_trimmed_second.fastq"
RNA3_TRIMMED_SECOND_DIR="ERR7132699_RNA_trimmed_second.fastq"
RNA4_TRIMMED_SECOND_DIR="ERR7132707_RNA_trimmed_second.fastq"
TRIMMOMATIC_CODE_DIR="Apps/Trimmomatic-0.39"
ILLUMINACLIP_ADAPTORS_DIR="Apps/Trimmomatic-0.39/adapters/TrueSeq_2AND3-SE.fa"

# DNA
# Quast - checking DNA quality
python "$QUAST_CODE_DIR/quast.py" $GENOME_FASTA_DIR -g $GENOME_ANNOTATION_DIR -o $QUAST_OUTPUT_DIR

# Genome Indexing
STAR --runMode genomeGenerate --genomeDir $GENOME_INDICES_OUTPUT_DIR --genomeFastaFiles $GENOME_FASTA_DIR --sjdbGTFfile $GENOME_ANNOTATION_DIR --sjdbOverhang 90

# RNA
# FastQC - checking RNA quality
fastqc $RNA1_DIR
fastqc $RNA2_DIR
fastqc $RNA3_DIR
fastqc $RNA4_DIR

# Filter the RNA to improve quality
java -jar "$TRIMMOMATIC_CODE_DIR/trimmomatic-0.39.jar" SE -phred33 $RNA1_DIR $RNA1_TRIMMED_FIRST_DIR ILLUMINACLIP:${ILLUMINACLIP_ADAPTORS_DIR}:2:30:10 HEADCROP:10
java -jar "$TRIMMOMATIC_CODE_DIR/trimmomatic-0.39.jar" SE -phred33 $RNA2_DIR $RNA2_TRIMMED_FIRST_DIR ILLUMINACLIP:${ILLUMINACLIP_ADAPTORS_DIR}:2:30:10 HEADCROP:10
java -jar "$TRIMMOMATIC_CODE_DIR/trimmomatic-0.39.jar" SE -phred33 $RNA3_DIR $RNA3_TRIMMED_FIRST_DIR ILLUMINACLIP:${ILLUMINACLIP_ADAPTORS_DIR}:2:30:10 HEADCROP:10
java -jar "$TRIMMOMATIC_CODE_DIR/trimmomatic-0.39.jar" SE -phred33 $RNA4_DIR $RNA4_TRIMMED_FIRST_DIR ILLUMINACLIP:${ILLUMINACLIP_ADAPTORS_DIR}:2:30:10 HEADCROP:10

# FastQC - checking RNA quality
fastqc $RNA1_TRIMMED_FIRST_DIR
fastqc $RNA2_TRIMMED_FIRST_DIR
fastqc $RNA3_TRIMMED_FIRST_DIR
fastqc $RNA4_TRIMMED_FIRST_DIR

# Filter the RNA to improve quality
java -jar "$TRIMMOMATIC_CODE_DIR/trimmomatic-0.39.jar" SE -phred33 $RNA1_TRIMMED_FIRST_DIR $RNA1_TRIMMED_SECOND_DIR SLIDINGWINDOW:4:15 MINLEN:36
java -jar "$TRIMMOMATIC_CODE_DIR/trimmomatic-0.39.jar" SE -phred33 $RNA2_TRIMMED_FIRST_DIR $RNA2_TRIMMED_SECOND_DIR SLIDINGWINDOW:4:15 MINLEN:36
java -jar "$TRIMMOMATIC_CODE_DIR/trimmomatic-0.39.jar" SE -phred33 $RNA3_TRIMMED_FIRST_DIR $RNA3_TRIMMED_SECOND_DIR SLIDINGWINDOW:4:15 MINLEN:36
java -jar "$TRIMMOMATIC_CODE_DIR/trimmomatic-0.39.jar" SE -phred33 $RNA4_TRIMMED_FIRST_DIR $RNA4_TRIMMED_SECOND_DIR SLIDINGWINDOW:4:15 MINLEN:36

# FastQC - checking RNA quality
fastqc $RNA1_TRIMMED_SECOND_DIR
fastqc $RNA2_TRIMMED_SECOND_DIR
fastqc $RNA3_TRIMMED_SECOND_DIR
fastqc $RNA4_TRIMMED_SECOND_DIR






